### ISSUE TITLE
redundant-clone false-positive for structopt attrs

### ISSUE THREAD

**Issue by MikailBag** [rust-lang/rust-clippy#4682]
For following code:

```rust

#[derive(StructOpt, Debug)]
struct ExecOpt {
//...
 #[structopt(
        short = "x",
        long = "expose",
        parse(try_from_str = parse_path_exposition_item)
    )]
    exposed_paths: Vec<minion::PathExpositionOptions>,
//...
}
```
Clippy gives error:
```
error: redundant clone
  --> src/minion-cli/src/main.rs:93:30
   |
93 |         parse(try_from_str = parse_path_exposition_item)
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::redundant-clone` implied by `-D clippy::all`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone
```
Upd: 
clippy 0.0.212 (5cb9833 2019-10-08)
rustc 1.40.0-nightly (c27f7568b 2019-10-13)

**Comment by sinkuu** [comment]
There would be actually [a redundant `.to_string()`](https://github.com/TeXitoi/structopt/blob/2cb1fbf67981950088ce4c1199c020fe52d68eb2/structopt-derive/src/lib.rs#L131) if your `try_from_str` function returns `Result<_, String>`. Anyway, try upgrading `structopt` to `0.3.3`, as they have added `#[allow(clippy)]` to the generated code.

**Comment by MikailBag** [comment]
Thanks for fast reply. Issue disappeared after upgrading to structopt v0.3
